### ISSUE TITLE
Generated files like AssemblyInfo.cs is processed by stylecop when using cli tool.

### ISSUE THREAD

**Issue by Looooopy** [DotNetAnalyzers/StyleCopAnalyzers#2314]
I get error SA1412 on AssemblyInfo.cs that are generated by MSBuild in the latest release of dotnet sdk when using the command prompt to build "dotnet build" but i do not get this error in Visual Studio 2017 when compiling there.

It complains on the AssemblyInfo.cs  in the obj folder.
I am using the "StyleCop.Analyzers 1.1.0-beta001".

My temporary solution was to set SA1412 to warning level but i don't think it should process the auto generated files, is there a way to turn this off or is it a bug?

**Comment by vweijsters** [comment]
This sounds like a bug, but it looks like a Roslyn bug. We rely on the Roslyn code to do the auto generated detection and as far as I can see the implementation of SA1412 is correct. Could you please report this issue in the [dotnet/roslyn](https://github.com/dotnet/roslyn) repository (and link it here, so that it can be tracked)?

**Comment by sharwell** [comment]
@Looooopy Can you include an example of the contents of the **AssemblyInfo.cs** file you are seeing? I can help you be more specific in your bug report.

**Comment by Sumo-MBryant** [comment]
I'm getting SA1633 for my **AssemblyInfo.cs** files, here is an example:

File: `obj\Debug\netstandard1.5\Foo.AssemblyInfo.cs`

Contents:

```csharp
//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by a tool.
//     Runtime Version:4.0.30319.42000
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------

using System;
using System.Reflection;

...

// Generated by the MSBuild WriteCodeFragment class.
```

**Comment by sharwell** [comment]
@Sumo-MBryant What version of StyleCop Analyzers are you using? Starting with the 1.1 release, this project doesn't do *anything* related to whether or not a file contains generated code. If a file is reporting diagnostics when it should be considered generated code, then the bug either lies in the tool that generated file, or in Roslyn's ability (heuristics) to detect what is considered generated code.

**Comment by andreacassioli** [comment]
Hi,
I am experiencing the very same issue. I have tried to include the auto-generated tag with no success.

**Comment by Sumo-MBryant** [comment]
I'm currently using 1.1.0-beta001

**Comment by samcragg** [comment]
I'm getting this issue (using the latest stable 1.02) but it looks like it's due to how MSBuild produces the files. As mentioned in https://github.com/Microsoft/msbuild/issues/1912, the problem appears to be because when built inside VS the auto-generated AssemblyInfo.cs starts with the comment:

    //------------------------------------------------------------------------------
    // <auto-generated>
    ...
    // </auto-generated>
    //------------------------------------------------------------------------------

However, when building from the command line via dotnet-build, the file only has this comment at the start:

    // Generated by the MSBuild WriteCodeFragment class.

i.e. the auto-generated tag is missing. So it looks like the issue is with the MSBuild task (not sure which one though)?

**Comment by sharwell** [comment]
@samcragg The source of the string is [here](https://github.com/Microsoft/msbuild/blob/355707088fe927563c327687cb3e09434253b977/src/Tasks/Resources/xlf/Strings.xlf#L2103), and it's actually written to the output [here](https://github.com/Microsoft/msbuild/blob/6851538897f5d7b08024a6d8435bc44be5869e53/src/Tasks/WriteCodeFragment.cs#L298).

**Comment by sharwell** [comment]
I filed a bug on the offending project so this can be resolved. 👍
